File: backend/core-api/src/main/java/ai/govbiz/core/service/AnalyzedSearchIntent.java
package ai.govbiz.core.service;

import java.util.List;

public record AnalyzedSearchIntent(
        List<String> keywords,
        List<String> regions,
        List<String> categories,
        List<String> targetTerms,
        boolean clarificationNeeded,
        String clarificationQuestion
) {
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/service/AnalyzedSearchIntent.kt
package ai.govbiz.core.service

data class AnalyzedSearchIntent(
    val keywords: List<String>,
    val regions: List<String>,
    val categories: List<String>,
    val targetTerms: List<String>,
    val clarificationNeeded: Boolean,
    val clarificationQuestion: String?,
)
